### ISSUE TITLE
training(1496): finalize the BC-warm-start versus RL-only launch and loader packet

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#8892]
## Archetype Metadata

```yaml
archetype: training-preparation
evidence_tier: preflight_valid
resource: local
execution_mode: unattended
one_pr: true
parallel_safe: true
linked_policy:
  - docs/benchmark_governance.md
  - docs/context/artifact_evidence_vocabulary.md
  - docs/maintainer_values.md
  - .agents/skills/goal-autopilot/SKILL.md
```

## Execution contract

```yaml
execution:
  owning_repo: ll7/robot_sf_ll7
  mutation_repos:
    - ll7/robot_sf_ll7
  route_required: local
  external_inputs: []
```

## Parent and authority

Child of #1496. Dataset collection remains owned by #6127 and its current execution operator. This issue prepares the downstream training comparison without requiring the nominal private dataset in tests and without running training.

## Goal / Problem

Create an exact, outcome-free launch and evaluation packet for the first oracle-imitation comparison: RL-only control versus behavior-cloning warm start, with optional PPO fine-tuning only when current #1496 explicitly requires it. The packet must consume the versioned #6127 dataset manifest, preserve train/validation/holdout separation, use matched training budgets, and separate sample-efficiency from final-performance estimands.

## Inputs

- current #1496 body/comments and parent programme #1489;
- current #6127/#1470/#1397 dataset, split, leakage, schema, and custody contracts;
- existing `scripts/training/pretrain_from_expert.py`, `scripts/training/train_ppo_with_pretrained_policy.py`, imitation environment contract, checkpoint/normalizer owners, and current config-first training infrastructure;
- current RL-only baseline configs and fair budget owner;
- current held-out evaluation, row accounting, checkpoint selection, artifact, and uncertainty owners;
- a small repository-owned synthetic dataset fixture for tests.

## Scope

In scope:

1. Define exact comparison arms, algorithm/config IDs, initialization policy, training seeds, environment-step or simulator-call budget, evaluation cadence, checkpoint retention, selected-checkpoint rule, and held-out evaluation identities.
2. Use pure BC or BC plus the currently approved PPO fine-tuning path. DAgger remains excluded unless a separate current data-collection/leakage contract explicitly authorizes it.
3. Define a versioned dataset-consumer manifest with expected dataset semantic ID/version/digest, split identities, schema, feature/action contracts, normalization, sample weights if already owned, and required quality fields.
4. Add a fail-closed loader gate using synthetic fixtures for valid manifest, missing shard, digest drift, schema mismatch, split overlap, holdout contamination, invalid/fallback rows, non-finite values, action mismatch, and duplicate sample identity.
5. Freeze matched budget and configuration-difference checks between RL-only and BC-warm-start arms.
6. Freeze sample-efficiency and final-performance estimands separately, with current uncertainty, aggregation, missingness, and result classifications.
7. Generate deterministic expected training-run/checkpoint/evaluation identities without executing training.
8. Add a bounded training-startup canary packet that performs dataset load, model construction, one or a few optimization updates on synthetic/disjoint data, checkpoint write/read, and evaluation startup only.
9. Produce exact private-ops launch templates, expected storage, monitoring fields, stop rules, and preservation contract.

Out of scope:

- collecting the nominal oracle dataset;
- full BC, PPO, or RL-only training;
- DAgger or new architecture/reward/observation sweeps;
- selecting budgets/checkpoints after held-out outcomes;
- changing dataset/split semantics;
- performance, dissertation, or publication claims.

## Affected files

Preferred bounded surface:

- issue-specific comparison/training configs under existing training owners;
- one versioned consumer/launch packet and analysis contract;
- focused loader, parity, canary, and identity tests;
- concise context note and private execution commands.

Maximum 16 files and 1,100 net new lines. Core planner, simulator, reward, dataset schema, and benchmark metric implementations remain read-only.

## Definition of Done

- [ ] RL-only and BC-warm-start arms have exact configs, seeds, budgets, checkpoint rules, and evaluation identities.
- [ ] Dataset semantic ID, digest, split, schema, feature/action, normalization, and quality requirements are explicit.
- [ ] Leakage, holdout contamination, missing/invalid samples, digest drift, schema/action mismatch, and duplicates fail before training.
- [ ] Matched-budget parity is mechanically checked.
- [ ] Sample-efficiency and final-performance analyses are separate and outcome-free.
- [ ] Expected run/checkpoint/evaluation identities are deterministic.
- [ ] The startup canary uses synthetic or disjoint data and proves loader/model/checkpoint/evaluation wiring only.
- [ ] Private operations can stage and admit the nominal comparison without inferring a field.
- [ ] No nominal dataset access, training, or claim change occurs.

## Validation / Testing

```bash
uv run python scripts/training/pretrain_from_expert.py --help
uv run python scripts/training/train_ppo_with_pretrained_policy.py --help
uv run pytest -q tests -k "imitation or warm_start or pretrain or dataset_manifest or leakage"
uv run python <issue-1496-consumer-validator> --check --fixture <synthetic-dataset-manifest> --format json
uv run python <issue-1496-identity-builder> --check --format json
uv run python <issue-1496-startup-canary> --check-only --format json
uv run ruff check <changed Python files>
uv run ruff format --check <changed Python files>
BASE_REF=origin/main scripts/dev/check_docs_proof_consistency_diff.sh
git diff --check
PR_READY_MODE=final BASE_REF=origin/main scripts/dev/pr_ready_check.sh
```

Resolve exact focused paths from current owners and record them in the PR. Required tests must not access private dataset or checkpoint bytes.

## Stop Conditions

Stop when #1496 does not freeze a fair baseline/budget/checkpoint rule, current dataset/split semantics conflict, the nominal consumer requires DAgger or new data collection, another issue/PR owns the packet, or accurate preparation needs reward/observation/architecture changes. Record the exact decision or implementation owner instead of guessing.

## Added Value / Success Metrics

- Nominal training can start immediately after dataset preservation and deadline admission.
- Leakage and unfair-budget failures are caught before expensive GPU allocation.
- Sample-efficiency and final-performance claims cannot be conflated.

## Project Metadata

```yaml
priority: 1
expected_duration_hours: 9
requires_domain_review: false
requires_compute_authority: false
requires_external_input: false
requires_user_interaction: false
```

## Blocked by

None for local packet preparation. Nominal training remains blocked until #6127 provides a valid durable dataset, this packet merges, the startup canary passes against the real staged dataset, and private admission proves deadline and preservation feasibility.

**Comment by ll7** [comment]
Autopilot implementation claim acquired for issue #8892.

- claim ref: `agent-claims/issue-8892` at `ccb33b864dcdea36897175d1c4d0468efa6f7f99`
- lane: local linked-worktree implementation
- planned worktree: `/home/luttkule/git/robot_sf_ll7.worktrees/issue-8892-bc-warm-start-packet-20260910`
- scope: outcome-free BC-warm-start versus RL-only launch/loader packet; no dataset collection, training, compute submission, or benchmark/publication claim

The claim remains until a covering PR reaches terminal delivery; if the work is abandoned, the claim will be released with an explicit handoff. A stale claim will be cleaned up only after checking for a live PR or recent owner note.
